os_net_config/dcb_config.py

```diff
@@ -369,12 +370,16 @@ def apply(self):
             # In case of mellanox nic, set the mstconfig and do fwreset
             # If the DCBX mode is already set to FW (0), ignore
             # performing mstconfig and mstfwreset.
-            if 'mlx' in cfg['driver'] and dcbx_mode != 0:
+            if (
+                'mlx' in cfg['driver'] and
+                not is_dcbx_mstconfig_enabled(cfg['device'], cfg['pci_addr'])
+            ):
                 mstconfig(cfg['device'], cfg['pci_addr'])
                 mstfwreset(cfg['device'], cfg['pci_addr'])
 
             # Set the mode to Firmware
-            dcb_config.set_dcbx(mode=0)
+            if dcbx_mode != 0:
+                dcb_config.set_dcbx(mode=0)
```

Comment:
The function set_dcbx() seems to be used with "mode=0" - is setting mode to "Host" allowed/valid ? If only two mode is valid - then can mode be bool ?